app/models/types/interval.rb

```diff
@@ -0,0 +1,43 @@
+class Interval
+
+  attr_reader :lt, :gt
+
+  # Code and code system are required (at minimum).
+  def initialize(lt, gt)
+    @lt = lt
+    @gt = gt
+  end
+
+  # Converts an object of this instance into a database friendly value.
+  def mongoize
+    [ lt, gt ]
+  end
+
+  class << self
+
+    # Get the object as it was stored in the database, and instantiate
+    # this custom class from it.
+    def demongoize(object)
```

Comment:
Same with `interval`, we should clarify the inputs to `demongoize` (in a comment is fine)

Reply:
Agreed.